daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -228,6 +228,8 @@ class DataProcessor(val ssrd: SchemaSetRuntimeData)
     var optThrown: Maybe[Throwable] = None
     try {
       try {
+        Assert.usageErrorUnless(state.dataInputStreamIsValid, "Attempted to use an invalid input source. This can happen due to our position in the input source being preadvanced by a failed parse beyond the length that we can backtrack")
```

Comment:
"preadvanced" sounds odd here. Is there maybe better way to say this? To me that sounds like we're moving forward in the input source but this is really a backtrack issue.